src/amo/client.js

```diff
@@ -1,10 +1,20 @@
-import makeClient from 'core/client/base';
+import createClient from 'core/client/base';
 
 // Initialize the tracking.
 import 'core/tracking';
 
-import routes from './routes';
+import App from './components/App';
 import sagas from './sagas';
 import createStore from './store';
 
-makeClient(routes, createStore, { sagas });
+createClient(createStore, { sagas }).then(({ renderApp }) => {
+  renderApp(App);
+
+  if (module.hot) {
+    module.hot.accept('./components/App', () => {
+      // eslint-disable-next-line global-require
+      const NextApp = require('./components/App').default;
+      renderApp(NextApp);
```

Comment:
I'm not sure if this is the culprit but when a module reloads, it unmounts and remounts *everything*. It didn't do that before. Is there a way to fix that? If not, I guess it's not a big deal.

Here is an example. Go to the collection list. Click on a collection. Click *Delete this collection* so that it asks you the prompt -- are you sure? Edit some other component, like `Collection`. After the reload, you can see that `ConfirmButton` got remounted which caused its state to get reset. This is not helpful for hot module debugging. This doesn't happen on the master branch.

Reply:
That's the recommended way now, but I've seen people configureing redux HMR for that reason. I'll try and see if my redux patch fixes that issue.

Reply:
The `ConfirmButton` might not be the best example because the `withUIState()` HOC explicitly resets the state on mount. I guess that's why it gets closed on code change.

Reply:
Resetting state on mount always felt wrong to me in `withUIState()` but I didn't have concrete reasons for why. This is a concrete reason. I filed https://github.com/mozilla/addons-frontend/issues/5764 to fix it.